src/expandableCalendar/WeekCalendar/presenter.spec.js

```diff
@@ -0,0 +1,359 @@
+const {UPDATE_SOURCES} = require('../commons');
+const {default: Presenter} = require('./presenter');
+
+describe('WeekCalendar presenter tests', () => {
+  const twoDaysSameWeek = {
+    date: '2021-01-01',
+    prevDate: '2021-01-02'
+  };
+
+  const twoDaysDifferentWeek = {
+    date: '2021-01-01',
+    prevDate: '2021-01-05'
+  };
+
+  const twoSomeDays = {
+    date: '2021-01-12',
+    prevDate: '2021-01-13'
+  };
+
+  const makeContext = (datesInWeek = twoDaysSameWeek, updateSource = UPDATE_SOURCES.WEEK_SCROLL) => {
+    return {
+      ...datesInWeek,
+      updateSource
+    };
+  };
+
+  const makeUUT = args => {
+    return new Presenter(args);
+  };
+
+  const makeProps = (current = '2021-01-01', context = makeContext(), firstDay = 0) => {
+    return {current, context, firstDay};
+  };
+
+  describe('Should Component Update', () => {
+    it('Expect component to not update when on same week', () => {
```

Comment:
I think this one will fail now...

Reply:
I removed the `sameWeek` check as you did in `master`